### PR TITLE
4337 Passkeys example app: Use the new `SafeModulesSetup` lib, add a CI job to verify the project builds

### PR DESCRIPTION
**Pull request by mmv08**
This PR:
- Fixes the broken compilation of the 4337 + passkey example app
- Adds a CI job that builds the app to verify that it can be compiled
- Update the app to support changes from https://github.com/safe-global/safe-modules/pull/223 and https://github.com/safe-global/safe-modules/pull/221. We don't have the 4337 package published, so the example app always uses the upstream version and has to be kept up-to-date

### CODE REVIEW

File: examples/safe-4337-passkeys/src/components/SafeCard.tsx [hunk 1]
```diff
@@ -24,7 +25,10 @@ function SafeCard({ passkey, provider }: { passkey: PasskeyLocalStorageFormat; p
       singleton: SAFE_SINGLETON_ADDRESS,
       fallbackHandler: SAFE_4337_MODULE_ADDRESS,
       signerFactory: WEBAUTHN_SIGNER_FACTORY_ADDRESS,
-      signerData: ethers.solidityPacked(['uint256', 'uint256'], [passkey.pubkeyCoordinates.x, passkey.pubkeyCoordinates.y]),
+      signerData: ethers.solidityPacked(
+        ['uint256', 'uint256', 'uint256'],
+        [passkey.pubkeyCoordinates.x, passkey.pubkeyCoordinates.y, WEBAUTHN_VERIFIER_ADDRESS],
+      ),
```
[the remark this answers is not in the document]
Comment (mmv08) [reply examples/safe-4337-passkeys/src/components/SafeCard.tsx:31]: Yes, wondering how it worked then :smile:

File: examples/safe-4337-passkeys/src/logic/safe.ts [hunk 2]
```diff
@@ -77,7 +78,10 @@ function getWebAuthnSignerContract(provider: ethers.JsonRpcProvider, address: st
  * @returns The signer address.
  */
 function getSignerAddressFromPubkeyCoords(x: string, y: string): string {
-  const deploymentCode = ethers.solidityPacked(['bytes', 'uint256', 'uint256'], [WebAuthSignerBytecode, x, y])
```
Comment (nlordell) [inline examples/safe-4337-passkeys/src/logic/safe.ts:80]: I _think_ there is a nicer way to do this with `new ethers.ContractFactory`:

```ts
const factory = new ethers.ContractFactory(WebAuthnSignerAbi, WebAuthnSignerBytecode);
const { data: deploymentCode } = factory.getDeployTransaction(x, y, WEBAUTHN_VERIFIER_ADDRESS);
```

It is fine as-is, if that doesn't work.

**Review by nlordell — APPROVED** [review]
Looks good. Just some small nits.

File: examples/safe-4337-passkeys/src/components/SafeCard.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nlordell) [reply examples/safe-4337-passkeys/src/components/SafeCard.tsx]: I think they are equivalent in this case, but the suggestion is slightly more correct.

File: examples/safe-4337-passkeys/src/logic/safe.ts (hunk 2 above)
Comment (mmv08) [reply examples/safe-4337-passkeys/src/logic/safe.ts:80]: Hmm, I'm not sure if it's nicer. While I agree it abstracts the lower-level details away, it feels a little bit too magic. Also, not everyone uses ethers, and it's much more likely that alternative applications have a method similar to `getCreate2Address` than `getDeployTransaction` that returns the `deploymentCode`, thus I believe it works better for an example app. What do you think?

**Pull request merged**